brainiak/utils/fmrisim.py

```diff
@@ -1006,18 +1387,163 @@ def mask_brain(volume,
     # You might get a warning but ignore it
     mask = ndimage.zoom(mask_raw, zoom_factor, order=2)
 
-    # Anything that is in the bottom 1% (presumably the exterior of the
-    # brain, make zero)
-    mask[mask < 0.01] = 0
+    # Any proportion that is below threshold (presumably the exterior of the
+    # brain), make zero
+    mask[mask < mask_threshold] = 0
+
+    # create the mask in 4d
+    mask = np.ones(volume.shape) * mask
+
+    return mask
 
-    # Mask out the non grey matter regions
-    brain = volume * mask
 
-    return brain
+def _noise_dict_update(noise_dict):
+    """
+    Update the noise dictionary parameters, in case any were missing
+
+    Parameters
+    ----------
+    noise_dict : dict
+
+    A dictionary specifying the types of noise in this experiment
+
+    Returns
+    -------
+    Updated dictionary
+
+    """
+
+    # Check what noise is in the dictionary and add if necessary. Numbers
+    # determine relative proportion of noise
+
+    if 'overall' not in noise_dict:
+        noise_dict['overall'] = 0.1
+    if 'motion_sigma' not in noise_dict:
+        noise_dict['motion_sigma'] = 0.1
+    if 'drift_sigma' not in noise_dict:
+        noise_dict['drift_sigma'] = 0.5
+    if 'auto_reg_sigma' not in noise_dict:
+        noise_dict['auto_reg_sigma'] = 1
+    if 'physiological_sigma' not in noise_dict:
+        noise_dict['physiological_sigma'] = 0.1
+    if 'system_sigma' not in noise_dict:
+        noise_dict['system_sigma'] = 1
+    if 'snr' not in noise_dict:
+        noise_dict['snr'] = 30
+    if 'spatial_sigma' not in noise_dict:
+        if noise_dict['overall'] == 0:
+            noise_dict['spatial_sigma'] = 0.015
+        else:
+            noise_dict['spatial_sigma'] = 0.015 / noise_dict['overall']
+
+    # Print the mixing of the noise
+    total_var = noise_dict['motion_sigma'] ** 2
+    total_var += noise_dict['drift_sigma'] ** 2
+    total_var += noise_dict['auto_reg_sigma'] ** 2
+    total_var += noise_dict['physiological_sigma'] ** 2
+    total_var += noise_dict['system_sigma'] ** 2
+
+    # print('Noise mixture\n')
+    # tmp = str(noise_dict['motion_sigma'] ** 2 / total_var * 100)[0:5]
+    # print('Task noise:\t\t\t\t' + tmp)
+    # tmp = str(noise_dict['drift_sigma'] ** 2 / total_var * 100)[0:5]
+    # print('Drift noise:\t\t\t' + tmp)
+    # tmp = str(noise_dict['auto_reg_sigma'] ** 2 / total_var * 100)[0:5]
+    # print('Autoregression noise:\t' + tmp)
+    # tmp = str(noise_dict['physiological_sigma'] ** 2 / total_var * 100)[0:5]
+    # print('Physiological noise:\t' + tmp)
+    # tmp = str(noise_dict['system_sigma'] ** 2 / total_var * 100)[0:5]
+    # print('System noise:\t\t\t' + tmp)
+
+    return noise_dict
+
+
+def generate_noise(dimensions,
+                   stimfunction,
+                   tr_duration,
+                   mask=None,
+                   noise_dict={'overall': 0.1},
```

Comment:
The default value is only evaluated once and shared between all calls:
https://docs.python.org/3/tutorial/controlflow.html#default-argument-values
You should set `None` as the default value and make the assignment in the function body.